org.lflang/src/org/lflang/generator/ts/TSReactorGenerator.kt

```diff
@@ -0,0 +1,715 @@
+    private fun getTargetType(state: StateVar): String = tsGenerator.getTargetTypeW(state)
+    private fun getTargetType(a: Action): String = tsGenerator.getTargetTypeW(a)
+    private fun getTargetType(p: Port): String = tsGenerator.getTargetTypeW(p)
+    private fun getTargetType(t: Type): String = tsGenerator.getTargetTypeW(t)
+    private fun generateVarRef(reference: VarRef): String = tsGenerator.generateVarRef(reference)
+
+    private fun getInitializerList(state: StateVar): List<String> =
+        tsGenerator.getInitializerListW(state)
+    private fun getInitializerList(param: Parameter): List<String> =
+        tsGenerator.getInitializerListW(param)
+    private fun getInitializerList(param: Parameter, i: Instantiation): List<String> =
+        tsGenerator.getInitializerListW(param, i)
+
+    private fun federationRTIProperties(): LinkedHashMap<String, Any> {
+        return tsGenerator.federationRTIPropertiesW()
+    }
+
+    // Initializer functions
+    private fun getTargetInitializer(state: StateVar): String {
+        return getInitializerList(state).joinToString(",")
+    }
+    private fun getTargetInitializerHelper(param: Parameter,
+                                           list: List<String>): String {
+        return if (list.size == 0) {
+            errorReporter.reportError(param, "Parameters must have a default value!")
+        } else if (list.size == 1) {
+            list[0]
+        } else {
+            list.joinToString(", ", "[", "]")
+        }
+    }
+    private fun getTargetInitializer(param: Parameter): String {
+        return getTargetInitializerHelper(param, getInitializerList(param))
+    }
+    private fun getTargetInitializer(param: Parameter, i: Instantiation): String {
+        return getTargetInitializerHelper(param, getInitializerList(param, i))
+    }
+    private fun initializeParameter(p: Parameter): String {
+        return """${p.name}: ${getTargetType(p)} = ${getTargetInitializer(p)}"""
+    }
+
+    /**
+     * Return a TS type for the specified action.
+     * If the type has not been specified, return
+     * "Present" which is the base type for Actions.
+     * @param action The action
+     * @return The TS type.
+     */
+    private fun getActionType(action: Action): String {
+        if (action.type !== null) {
+            return getTargetType(action.type)
+        } else {
+            return "Present"
+        }
+    }
+
+    /**
+     * Return a TS type for the specified port.
+     * If the type has not been specified, return
+     * "Present" which is the base type for ports.
+     * @param port The port
+     * @return The TS type.
+     */
+    private fun getPortType(port: Port): String {
+        if (port.type !== null) {
+            return getTargetType(port.type)
+        } else {
+            return "Present"
+        }
+    }
+
+    private fun generateArg(v: VarRef): String {
+        return if (v.container !== null) {
```

Comment:
Using `!==` or `===` is not idiomatic in kotlin. This causes compiler warnings because you sometimes do it with primitive types (`code === 0`), which causes boxing before the comparison. Identity comparison of boxed primitives is deprecated because of planned future extensions to the JVM type system.

Sticking to  `!=` and `==` is better practice.

Reply:
Thanks, done.